3542/1/lib/gems.rb

```diff
@@ -0,0 +1,56 @@
+require 'yaml'
+require 'terminal-table'
+require_relative 'repo'
+
+class Gems
+  def initialize(params)
+    @params = params.map { |param| param.split('=') }.to_h
+
+    @gem_list = @params['--name'] ? filter_by_name(gem_list) : gem_list['gems']
+
+    @repos = @params['--top'] ? top(gems) : gems
+  end
+
+  #:reek:FeatureEnvy
+  # rubocop:disable Style/UnneededInterpolation
+  # rubocop:disable Metrics/MethodLength
+  def show
+    table_rows = @repos.map do |repo|
```

Comment:
> split method into couple method to resolve `# rubocop:disable Metrics/MethodLength`

@longbeardmen it's impossible, because if i carry out `table_rows` calculating - it's still will be long, because `table_rows` calculating take 11 lines.